src/platform_impl/linux/x11/mod.rs

```diff
@@ -252,48 +309,29 @@ impl<T: 'static> EventLoop<T> {
             calloop::Mode::Level,
         );
         handle
-            .insert_source(source, |_, _, _| Ok(calloop::PostAction::Continue))
+            .insert_source(source, |readiness, _, state| {
+                state.x11_readiness = readiness;
+                Ok(calloop::PostAction::Continue)
+            })
             .expect("Failed to register the X11 event dispatcher");
 
-        // Create a channel for sending user events.
-        let (user_sender, user_channel) = channel();
-        handle
-            .insert_source(user_channel, |ev, _, state| {
-                if let ChanResult::Msg(user) = ev {
-                    state.user_events.push_back(user);
-                }
+        let (waker, waker_source) =
+            calloop::ping::make_ping().expect("Failed to create event loop waker");
```

Comment:
If we must use the new channel design, can we switch to using [`LoopSignal`](https://docs.rs/calloop/latest/calloop/struct.LoopSignal.html) instead here?

Reply:
It looks like `LoopSignal` is really geared towards code that uses calloops `EventLoop::run()` API which we don't use